lib/logstash/outputs/s3/uploader.rb

```diff
@@ -44,15 +46,22 @@ def upload(file, options = {})
             #
             # Thread might be stuck here, but I think its better than losing anything
             # its either a transient errors or something bad really happened.
-            logger.error("Uploading failed, retrying.", :exception => e.class, :message => e.message, :path => file.path, :backtrace => e.backtrace)
-            sleep TIME_BEFORE_RETRYING_SECONDS
-            retry
+            if tries < @retry_count
+              tries += 1
+              logger.warn("Uploading failed, retrying (#{tries}).", :exception => e.class, :message => e.message, :path => file.path, :backtrace => e.backtrace)
```

Comment:
as retry logging seems to "annoy" users I decreased logger.level to **warn**
(and added a new error message when retrying is done)
but users not configuring `retry_count` will only see logging downgraded to warn level.
if that is a potential problem and we should keep error level here, I am fine to do that ...